robot/impl/local_robot.go

```diff
@@ -1203,6 +1210,7 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	// Cleanup unused packages after all old resources have been closed above. This ensures
 	// processes are shutdown before any files are deleted they are using.
 	allErrs = multierr.Combine(allErrs, r.packageManager.Cleanup(ctx))
+	allErrs = multierr.Combine(allErrs, r.localPackages.Cleanup(ctx))
```

Comment:
Unless I missed some tricky if statement, I think this is a merge-blocker. (very possible that I missed something)

Each package manager contains its own package map. The local_package_manager contains only local modules with synthetic packages

The commonCleanup will delete all files & folders that it does not know about / expect 

localPackages does not know about the other packages (because they are not added to its map)

This means that both of these package managers will delete each others packages.

With the way the code is structured, I do not know of a good way to fix this. Open to brainstorming

Reply:
(Only solution I can think of is to go back to having the localRobot contain only one package manager and then do more multiplexing inside the cloud package manager and local_package_manager... neither of which brings me joy)

Reply:
I like how we've entered 'sparks joy' territory for this PR

Reply:
cleanup thing is handled here
https://github.com/viamrobotics/rdk/pull/3921/files#diff-f9c3be8eaea3a73a6f911325ff37778b4811e89905c63cf4fb2e61b874805e04R69-R73

the packages dir for localPackageManager has a different path
```go
// LocalPackagesDir transforms a packagesDir string to the suffixed version for localManager.
// local + cloud manager need separate parent dirs so they don't delete each other in Cleanup.
func LocalPackagesDir(packagesDir string) string {
	return filepath.Clean(packagesDir) + "-local"
}
```

Reply:
Ahhh -- that is great news! I must admit that `LocalPackagesDir` is not the most obvious name--But neither are the others `LocalXYZDir` functions! 

I have checked and I agree that I can't find any code using the non `-local` suffixed package dir.

That crosses off my biggest concern.